### PR TITLE
Update upgrading guide after deprecation of `Provider#forUseAtConfigurationTime`

### PR DESCRIPTION
**Pull request by bamboo**
Followup to #18722

### CODE REVIEW

**Comment by bamboo** [comment]
@bot-gradle test ACC

**Review by eskatos — APPROVED** [review]
:+1:

**Comment by bamboo** [comment]
@bot-gradle test QFL

**Comment by vlsi** [comment]
This ia really sad. There is "a lot" of plugins that incorporated the use of Provider#forUseAtConfigurationTime, and you start nagging the users.

Have you considered to keep the methods as no-op deprecated one?

Removing the method in 8.0 does not seem to make Gradle maintenance easier, yet it adds overhead on plugin authors and gradle users.

**Comment by bamboo** [comment]
@bot-gradle test and merge

File: subprojects/model-core/src/main/java/org/gradle/api/internal/provider/AbstractMinimalProvider.java [hunk 1]
```diff
@@ -116,7 +116,14 @@ public T getOrElse(T defaultValue) {
...
     public final Provider<T> forUseAtConfigurationTime() {
-        // TODO:configuration-cache nag user
+        /*
+ TODO:configuration-cache start nagging in Gradle 8.x
+        DeprecationLogger.deprecateMethod(Provider.class, "forUseAtConfigurationTime")
+            .withAdvice("Simply remove the call.")
+            .willBeRemovedInGradle9()
+            .withUpgradeGuideSection(7, "for_use_at_configuration_time_deprecation")
+            .nagUser();
+*/
```
Comment (eskatos) [inline subprojects/model-core/src/main/java/org/gradle/api/internal/provider/AbstractMinimalProvider.java:126]: :x: I would not leave the code commented but create an issue assigned to 8.0 instead

**Pull request merged**